src/common/calendar/date/CalendarEventsRepository.ts

```diff
@@ -355,17 +356,7 @@ export class CalendarEventsRepository {
 		const uid = generateUid(calendarId, Date.now())
 
 		const eventTitle = this.calendarModel.getBirthdayEventTitle(contact.firstName)
-
-		let fullDateIso = contact.birthdayIso!
-		// Set the year because we can have birthdays without year
-		if (contact.birthdayIso?.startsWith("--")) {
-			fullDateIso = contact.birthdayIso.replace("-", "1970")
-		}
-
-		// Set up start and end date base on UTC.
-		// Also increments a copy of startDate by one day and set it as endDate
-		const startDate = new Date(fullDateIso!) // Uses fullDateIso as UTC value, therefore it creates a new Date applying the local timezone on top of fullDateIso
-		const endDate = incrementDate(new Date(startDate), 1)
+		const { startDate, endDate } = getAllDayDatesUTCFromIso(contact.birthdayIso!, this.zone)
```

Comment:
Are we are actually sure that we have a birthdayIso field? What happens in that case?